ra_aid/file_listing.py

```
@@ -0,0 +1,131 @@
+        
+        # Get total count before truncation
+        total_count = len(files)
+        
+        # Truncate if limit specified
+        if limit is not None:
+            files = files[:limit]
+            
+        return files, total_count
+
+    except subprocess.CalledProcessError as e:
+        raise GitCommandError(f"Git command failed: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)